sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -465,7 +498,7 @@ case class ParquetRelation2
     // before calling execute().
 
     val job = new Job(sqlContext.sparkContext.hadoopConfiguration)
-    val writeSupport = if (schema.map(_.dataType).forall(_.isPrimitive)) {
+    val writeSupport = if (parquetSchema.map(_.dataType).forall(_.isPrimitive)) {
```

Comment:
For `SaveMode.Overwrite`, we are always writing to an empty directory, so there's no "existing one".